### ISSUE TITLE
Qhelpgenerator debug mode

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9135]
**Describe the bug**
Not a bug.

**Expected behavior**
It would be nice if, when `-d qhp` is enabled, qhelpgenerator is run first (twice) with the flags `-v -c`, which print the version and check that all files from the file list are found. (I think they cannot be used simultaneously, nor with -o).

**To Reproduce**
See the reproducer at https://github.com/doxygen/doxygen/issues/9094

**Screenshots**
```
Usage:

qhelpgenerator <help-project-file> [options]

  -o <compressed-file>   Generates a Qt compressed help
                         file called <compressed-file>.
                         If this option is not specified
                         a default name will be used.
  -c                     Checks whether all links in HTML files
                         point to files in this help project.
  -v                     Displays the version of 
                         qhelpgenerator.
```

**Version**
1.9.4 (15dae504a167d838873053cc96417d8beae78df7*)

**Additional context**
Included HTML path problems found in https://github.com/doxygen/doxygen/issues/9094

**Comment by albert-github** [comment]
As is the idea is sympathetic (and not difficult to implement) though I tried to run the `-c` option on the mentioned case and it takes ages (runs already over 20 minutes at my place).

The documentation states: 
> Checks whether all links in HTML files point to files in this help project.

I think one can either interpret this as:
- checks whether all files mentioned in the `index.qhp` exist 

or
- checks whether all files mentioned in the `index.qhp` exist **and** all links in the mentioned files exist.

It looks to me that the later is done here. 

So I don't think it is a good idea to run the `-c` even under debug conditions.

Regarding the `-v` I'm not sure this is something doxygen should output either, though with the `-d qhp` this would be easy again.

- how fast did you see the results of the `-c`?
- @doxygen what to do?

Did you miss some files in the example besides the files in  the "user" : of the layout files (i.e. "group__Tutorials.html" and "release-notes.html")

**Comment by albert-github** [comment]
I tried:
- 6.1.2 and this has the behavior as indicated
  -  did also do the test without the `index.html` and in that case it quickly shows  that the `index.html` is missing and after this it looks like it is in an end less loop or doing some very long lasting operations
- 6.2.2 ant this also takes ages
- 5.15.2 runs fast though
  -  as you mentioned doesn't warn at all for the groupTutorials and release-notes, so looks indeed like it is not looking at the `<toc>` part. 

So I'm still think it is not wise to enable the `-c` flag in debug mode.

**Comment by ferdymercury** [comment]
Hmmm looks that the newer qhelpgenerator is buggy then. This is a small project file, so even when recursing links, it should finish in a couple of seconds. Agreed, let's skip this then. Or we can leave it open for the moment and submit a bug report to Qt.

**Comment by albert-github** [comment]
Best would indeed that you submit a bug report to Qt (and post a link here).
Might be that they could also explain / enhance a bit about what is checked.

**Comment by ferdymercury** [comment]
Good point, thanks. Here the link: https://bugreports.qt.io/browse/QTBUG-101070

**Comment by ferdymercury** [comment]
A patch has been proposed https://codereview.qt-project.org/c/qt/qttools/+/400691

**Comment by ferdymercury** [comment]
Patch has been released for 6.4.0 Alpha

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9228

**Comment by ferdymercury** [comment]
Thanks!!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
